hypothesis_geometry/planar.py

```diff
@@ -162,12 +167,82 @@ def triangular_contours(x_coordinates: Strategy[Coordinate],
             .map(list))
 
 
+def segments(x_coordinates: Strategy[Coordinate],
+             y_coordinates: Optional[Strategy[Coordinate]] = None
+             ) -> Strategy[Segment]:
+    """
+    Returns a strategy for segments.
+
+    :param x_coordinates: strategy for endpoints' x-coordinates.
+    :param y_coordinates:
+        strategy for endpoints' y-coordinates,
+        ``None`` for reusing x-coordinates strategy.
+    """
+
+    def non_degenerate_segment(segment: Segment) -> bool:
+        start, end = segment
+        return start != end
+
+    points_strategy = points(x_coordinates, y_coordinates)
+    return (strategies.tuples(points_strategy, points_strategy)
+            .filter(non_degenerate_segment))
+
+
+def polylines(x_coordinates: Strategy[Coordinate],
+              y_coordinates: Optional[Strategy[Coordinate]] = None,
+              *,
+              min_size: int = MIN_POLYLINE_SIZE,
+              max_size: Optional[int] = None) -> Strategy[Polyline]:
+    """
+    Returns a strategy for polylines.
+
+    :param x_coordinates: strategy for vertices' x-coordinates.
+    :param y_coordinates:
+        strategy for vertices' y-coordinates,
+        ``None`` for reusing x-coordinates strategy.
+    :param min_size: lower bound for polyline size.
+    :param max_size: upper bound for polyline size, ``None`` for unbound.
+    """
+    _validate_sizes(min_size, max_size, MIN_POLYLINE_SIZE)
+    result = _polylines(x_coordinates, y_coordinates, min_size, max_size)
+
+    if max_size is None or max_size > MIN_POLYLINE_SIZE:
+        def close(polyline: Polyline) -> Polyline:
+            return polyline + [polyline[0]]
+
+        result |= (_polylines(x_coordinates, y_coordinates,
+                              min_size - 1
+                              if min_size > MIN_POLYLINE_SIZE
+                              else min_size,
+                              max_size - 1
+                              if max_size is not None
+                              else max_size)
+                   .map(close))
```

Comment:
This part is confusing. How about adding a comment?

Reply:
renamed function to `close_polyline`

Reply:
Sorry, I meant this part:
```
min_size - 1
if min_size > MIN_POLYLINE_SIZE
else min_size,
max_size - 1
if max_size is not None
else max_size
```
not the function name.

Reply:
done